crates/vsock/src/main.rs

```diff
@@ -209,7 +211,18 @@ impl TryFrom<VsockArgs> for Vec<VsockConfig> {
                     )])
                 }),
             },
+        }?;
+
+        // check if there are duplicate CIDs
+        let mut cid_set: HashSet<u64> = HashSet::new();
```

Comment:
Instead of checking here, what about doing it when we create the backend/thread in `VhostUserVsockThread::new()` https://github.com/rust-vmm/vhost-device/blob/2fa80555d242f5aaa1d188cf79f982263c8bfd51/crates/vsock/src/vhu_vsock_thread.rs#L114C41-L114C41?

We should have less code to add, and it could also help us in the future if we manage new VMs at runtime.

Reply:
Could be done there as well. I just didn't want to spawn a new thread (with possibly the same name as another one) when I could check for conflicts beforehand. What do you think?

Reply:
I'm not very convinced, I would manage the CIDs all in one place. Using a set here just to check for duplicates and then inserting the same CID in a map seems to me to split the management in 2 places and in the future may not be easily maintainable.

I don't know if adding the thread in the map in `VhostUserVsockThread::new()` is the best thing, but I would like to centralize adding and checking duplicates in one place. (Feel free to move everything somewhere else if you think it would be better).

Also, another thing I don't like is controlling it in configuration parsing. There we should just check that the parameters are okay, but the validation should be done later. Like for example we do for paths: we don't check, while parsing the options, whether we have permissions to access the path or not.

Reply:
Ok, I see your point. I will change the implementation to your suggested approach.

Reply:
Done!